spydra/src/main/java/com/spotify/spydra/submitter/api/DynamicSubmitter.java

```diff
@@ -213,9 +212,13 @@ private void waitForHistoryToBeMoved(SpydraArgument arguments) throws IOExceptio
     String path = arguments.clusterProperties()
         .getProperty("mapred:mapreduce.jobhistory.intermediate-done-dir");
     URI uri = URI.create(path);
-    FileSystem fs = gcpUtils.fileSystemForUri(uri);
+    String bucketName = uri.getHost();
+    String directory = uri.getPath();
+    if (directory != null && directory.length() > 0) {
+      directory = directory.substring(1, directory.length()); //remove trailing slash from path
+    }
     LOGGER.info("Waiting for history files to be moved to its final location");
-    while (fs.getContentSummary(new Path(uri)).getFileCount() != 0) {
+    while (new StorageUtils().getFileCount(bucketName, directory+"/") != 0) {
```

Comment:
I would use something else than file count here, just to avoid the lazy paging happening in the getFileCount method.